docs/using-hive.md

```diff
@@ -1263,38 +1267,82 @@ If the cluster you are looking to adopt is on AWS and leverages Privatelink, you
-
-```yaml
-  clusterMetadata:
-    adminKubeconfigSecretRef:
-      name: my-gcp-cluster-admin-kubeconfig
-    clusterID: 61010205-c91d-44c9-8394-3e1790bd76f3
-    infraID: my-gcp-cluster-wsvdn
-    platform:
-      gcp:
-        networkProjectID: some@project.id
+metadata.json sample for GCP cluster
+
+```
+{
+  "clusterName": "my-gcp-cluster",
+  "clusterID": "61010205-c91d-44c9-8394-3e1790bd76f3",
+  "infraID": "my-gcp-cluster-wsvdn",
+  "gcp": {
+    "region": "us-east1",
+    "projectID": "projectID for the cluster",
+    "networkProjectID": "networkProjectID for the cluster",
+    "privateZoneDomain": "privateZoneDomain like <clusterName>.<baseDomain>",
+    "privateZoneProjectID": "privateZoneProjectID for the cluster"
+  },
+  "featureSet": "featureSet for the cluster",
+  "customFeatureSet": null
+}
+```
+
+Note: networkProjectID is optional and shall be included if the cluster you are looking to adopt is on GCP and uses a shared VPC
+
+metadata.json sample for AWS cluster
+
+```
+{
+  "clusterName": "my-aws-cluster",
+  "clusterID": "61010205-c91d-44c9-8394-3e1790bd76f3",
+  "infraID": "my-aws-cluster-wsvdn",
+  "aws": {
+    "region": "us-east-1",
+    "identifier": [
+      {
+        "kubernetes.io/cluster/<infraID>": "owned"
+      },
+      {
+        "openshiftClusterID": "<clusterID>"
+      },
+      {
+        "sigs.k8s.io/cluster-api-provider-aws/cluster/<infraID>": "owned"
+      }
+    ],
+    "clusterDomain": "<clusterName>.<baseDomain>",
+    "hostedZoneRole": "arn:aws:iam::641733028092:role/<clusterName>-shared-role"
+  },
+  "featureSet": "featureSet for the cluster",
+  "customFeatureSet": null
+}
+```
+
+Note: hostedZoneRole is optional and shall be included if the cluster you are looking to adopt is on AWS and uses a shared VPC
+
+metadata.json sample for Azure cluster
+
+```
+{
+  "clusterName": "my-azure-cluster",
+  "clusterID": "61010205-c91d-44c9-8394-3e1790bd76f3",
+  "infraID": "my-azure-cluster-wsvdn",
+  "azure": {
+    "armEndpoint": "",
+    "cloudName": "AzurePublicCloud",
+    "region": "centralus",
+    "resourceGroupName": "",
+    "baseDomainResourceGroupName": "baseDomainResourceGroupName for the cluster"
+  },
+  "featureSet": "featureSet for the cluster",
+  "customFeatureSet": null
+}
 ```
 
 ### Adopting with hiveutil
 
 [hiveutil](hiveutil.md) is a development focused CLI tool which can be built from the hive repo. To adopt a cluster specify the following flags:
 
 ```bash
-bin/hiveutil create-cluster --namespace=namespace-to-adopt-into --base-domain=example.com mycluster --adopt --adopt-admin-kubeconfig=/path/to/cluster/admin/kubeconfig --adopt-infra-id=[INFRAID] --adopt-cluster-id=[CLUSTERID]
+bin/hiveutil create-cluster --namespace=namespace-to-adopt-into --base-domain=example.com mycluster --adopt --adopt-admin-kubeconfig=/path/to/cluster/admin/kubeconfig --adopt-infra-id=[INFRAID] --adopt-cluster-id=[CLUSTERID] --adopt-metadata-json=/path/to/metadata.json
```

Comment:
This is cool and we should keep it. However, we should also include instructions for creating the Secret from a file directly, above, for customers who are well behaved and actually read the admonishment that hiveutil is not supposed to be for customer use :)